tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -546,7 +550,12 @@ void test_multiple_alarms_instance(const char *dev_name)
 	err = counter_set_channel_alarm(dev, 1, &alarm_cfg2);
 	zassert_equal(0, err, "%s: Counter set alarm failed", dev_name);
 
-	k_busy_wait(1.2*counter_ticks_to_us(dev, ticks * 2U));
+#ifdef CONFIG_COUNTER_MCUX_CTIMER
+	k_busy_wait((uint32_t)counter_ticks_to_us(dev, 0xFFFFFFFF));
```

Comment:
this looks strange. If it's because ctimer does not support setting top value then i would just skip this test for that counter. You can do this by adding `set_top_value_capable()` to `multiple_channel_alarm_capable` check.

Reply:
That feeds back into one of my earlier comments, the ctimers could support this behaviour by making use of one of the channels to provide the resetting behaviour that setting the top value provides. This could be done by reducing the number of user usable channels from 4 to 3, reserving a channel for detecting the set top value and resetting or if the user sets a top value using one of the channels and producing an error if the user wants to use that channel as well as setting a top value.

Or I could just leave the driver as is and exclude the test as you suggest.